074-connector-offsets-support.md

```diff
@@ -0,0 +1,226 @@
+* Skipping records that cause issues with a connector and cannot be addressed using existing error-handling features
+
+## Proposal
+
+Strimzi should include mechanisms to let users manage offsets when utilising the KafkaConnector resources.
+This proposal suggests different approaches for each action, since they are all slightly different.
+In summary:
+ - listing offsets via a ConfigMap and enabled/disabled through an annotation
+ - altering offsets via an annotation and ConfigMap
+ - deleting offsets via an annotation on the KafkaConnector resource and through a configuration option when the connector is deleted
+
+### Request/Response from the API endpoints
+
+The response body received when using the GET endpoint, and the request body used when calling the PATCH endpoint have identical structures to allow users to easily fetch and then alter offsets.
+The structure of the body is different for source and sink connectors.
+This is because sink connectors use regular Kafka offsets, whereas source connectors determine their own structure for offsets.
+This difference is reflected in the approaches outlined in this proposal.
+
+Source connector format is:
+```
+{
+  "offsets": [
+    {
+      "partition": {
+        // Connector-defined source partition
+      },
+      "offset": {
+        // Connector-defined source offset
+      }
+    }
+  ]
+}
+```
+
+Sink connector format is:
+```
+{
+  "offsets": [
+    {
+      "partition": {
+        "kafka_topic": // Kafka topic
+        "kafka_partition": // Kafka partition
+      },
+      "offset": {
+        "kafka_offset": // Kafka offset
+      }
+    }
+  ]
+}
+```
+
+### New annotation
+
+All three actions will make use of a new annotation called `strimzi.io/connectorOffsets`.
+The possible values will be `list`, `alter` and `reset`.
+
+### Listing offsets
+
+To list the current offsets the user will annotate the KafkaConnector resource with the new annotation `strimzi.io/connectorOffsets` set to `list`.
+After the annotation is added, on the next reconciliation the operator will fetch the current offsets and create a Kubernetes ConfigMap containing the JSON response.
+If the ConfigMap already exists the operator will replace the data with the updated data.
+The operator will then remove the annotation from the KafkaConnector CR.
+
+If the user wants to see an updated set of offsets they will need to re-annotate the resource.
+
+The ConfigMap the operator creates will be called `{connector}-offsets`, where `{connector}` is the name of the KafkaConnector CR that was annotated.
+The ConfigMap will contain a single file called `offsets.json` that contains the response received from the GET /connectors/{connector}/offsets endpoint.
+The KafkaConnector CRD will also be updated to contain a new property called `listOffsetsCM` that when set will override the name of the ConfigMap the operator creates/updates with the endpoint response.
```

Comment:
Not sure I would use the `CM` to specify it's a ConfigMap (it would come from the CRD spec). Should it be `offsetsList` as well?